Automattic/simplenote-ios
# Spring Cleaning: Checklist Button

### Fix

This PR is to address the checklist button state. Design calls for button to show/hide based on focused state of note. Could also just disable if we prefer that behaviour.

Currently the button is always present and clicking it without a focused note (ie: ready for edits) inserts a checklist item at what feels like a random location. Most likely this is insertion point as tracked by the UITextView. Side effect is the note is marked as modified and triggers a sync.

If the insertion point is below the fold, it isn't clear the note has even been modified.

https://d.pr/i/Zy3exj — Having trouble uploading/embedding the GIF.

### Test

1. Open a note but do not click to activate the keyboard. Verify the Checklist button is not visible in the navigation.
2. Tap on the note to focus it for editing. The keyboard should appear and the Checklist button should become visible.
3. Tap the Keyboard button to dismiss the keyboard. The Checklist button should also disappear.

### Test with Hardware Keyboard

This one is harder to do in the simulator but you can do it by toggling on the hardware keyboard option. Select **Keyboard > Connect Hardware Keyboard** from the **I/O** menu then repeating the above steps. Verify that the Checklist button only appears when the note has focus, regardless of keyboard visibility.

### Review

cc @jleandroperez — Would you mind having a look at this please?!

### Release

The Checklist button is now shown only when a note is focused for editing.
